perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -589,31 +640,57 @@ def _extract_vm_metrics(time_intervals_list, folders, mount_type) -> list:
       directory_structure.folders, persistent_disk_time_intervals, args.message[0],
       int(args.num_samples[0]))
 
-  if args.upload:
-    log.info('Uploading files to the Google Sheet.\n')
-    _export_to_gsheet(
-        directory_structure.folders, gcs_parsed_metrics, args.command[0],
-        WORKSHEET_NAME_GCS)
-    _export_to_gsheet(
-        directory_structure.folders, pd_parsed_metrics, args.command[0],
-        WORKSHEET_NAME_PD)
-
   print('Waiting for 360 seconds for metrics to be updated on VM...')
   # It takes up to 240 seconds for sampled data to be visible on the VM metrics graph
   # So, waiting for 360 seconds to ensure the returned metrics are not empty.
   # Intermittently custom metrics are not available after 240 seconds, hence
   # waiting for 360 secs instead of 240 secs
   time.sleep(360)
 
-  gcs_vm_results = _extract_vm_metrics(gcs_bucket_time_intervals, directory_structure.folders, 'gcs_bucket')
-  pd_vm_results = _extract_vm_metrics(persistent_disk_time_intervals, directory_structure.folders, 'persistent_disk')
+  gcs_results = _extract_vm_metrics(gcs_bucket_time_intervals, directory_structure.folders, 'gcs_bucket')
+  pd_results = _extract_vm_metrics(persistent_disk_time_intervals, directory_structure.folders, 'persistent_disk')
 
   for folder in directory_structure.folders:
     print(f'VM metrics for listing tests (gcs bucket) for folder: {folder.name}...')
-    print(gcs_vm_results[folder.name])
+    print(gcs_results[folder.name])
+
+  gcs_results_vm = {}
+  pd_results_vm = {}
+
+  for folder in directory_structure.folders:
+    gcs_results_vm[folder.name] = gcs_results[folder.name][2:]
+    pd_results_vm[folder.name] = pd_results[folder.name][2:]
+
+  temp_results_gcs = _get_values_to_export(directory_structure.folders, gcs_parsed_metrics, args.command[0])
+  temp_results_pd = _get_values_to_export(directory_structure.folders, pd_parsed_metrics, args.command[0])
+
+  results_gcs = []
+  results_pd = []
+
+  for folder, values_gcs, values_pd in zip(directory_structure.folders, temp_results_gcs, temp_results_pd):
+    # Only some extracted metrics will be uploaded to Google Spreadsheets and BigQuery
+    upload_values_gcs = values_gcs[1:3] + [values_gcs[4]] + [values_gcs[8]] + [values_gcs[17]] + values_gcs[5:8] + values_gcs[9:13]
```

Comment:
can we make changes in _get_values_to_export function to return only those metrics that we need to upload ? https://github.com/GoogleCloudPlatform/gcsfuse/blob/23e88afb5fb53a01de51ff74ea9b75791837473f/perfmetrics/scripts/ls_metrics/listing_benchmark.py#L84.
So that we don't have to select values by index here ?

Reply:
Sure, that's possible.